python/ray/_private/node.py

```diff
@@ -1498,5 +1498,6 @@ def _remove_protocol_from_url(self, url: Optional[str]) -> str:
         parsed_url = urllib.parse.urlparse(url)
         if parsed_url.scheme:
             # Construct URL without protocol
-            return parsed_url.netloc + parsed_url.path
+            scheme = "%s://" % parsed_url.scheme
+            return parsed_url.geturl().replace(scheme, "", 1)
```

Comment:
Would it make more sense to return `parsed_url.geturl()` of which we stripped `parsed_url.netloc` from the left? Seems possibly more robust to me :)

Reply:
I was trying to do something like that, but it doesn't seem to be working as expected
```
import urllib
url = "https://dashboard_url/path/?token=token1"
parsed_url = urllib.parse.urlparse(url)

parsed_url
>>> ParseResult(scheme='https', netloc='dashboard_url', path='/path/', params='', query='token=token1', fragment='')
parsed_url._replace(scheme="")
>>> ParseResult(scheme='', netloc='dashboard_url', path='/path/', params='', query='token=token1', fragment='')
parsed_url._replace(scheme="").geturl()
>>> '//dashboard_url/path/?token=token1'
```

Reply:
I see, ok sounds good I guess you are more or less already doing that :)